js/modules/k6/http/ntlm.go

```diff
@@ -0,0 +1,132 @@
+/*
+ *
+ * k6 - a next-generation load testing tool
+ * Copyright (C) 2018 Load Impact
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as
+ * published by the Free Software Foundation, either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+package http
+
+import (
+	"bytes"
+	"encoding/base64"
+	"io"
+	"io/ioutil"
+	"net/http"
+	"strings"
+
+	"github.com/ThomsonReutersEikon/go-ntlm/ntlm"
+	"github.com/pkg/errors"
+)
+
+// NTLMNegotiator is a http.Roundtripper that automatically converts basic authentication to
+// NTLM authentication.
+type NTLMNegotiator struct{ http.RoundTripper }
+
+func (n NTLMNegotiator) RoundTrip(req *http.Request) (res *http.Response, err error) {
+	// Use default round tripper if not provided
+	rt := n.RoundTripper
+	if rt == nil {
+		rt = http.DefaultTransport
```

Comment:
I think we should return an error here instead of using the default transport, since I don't thing we would ever want to use it in `k6` - it doesn't have `netext.Dialer` and the user-supplied configuration